homeassistant/components/kodi.py

```diff
@@ -0,0 +1,77 @@
+"""
+components.kodi
+~~~~~~~~~~~~~~~~~~
+Component that handles communication with a Kodi media player device.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/kodi/
+"""
+import logging
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import extract_domain_configs
+from homeassistant.util import get_random_string
+
+DOMAIN = "kodi"
+REQUIREMENTS = ['jsonrpc-requests==0.1']
+_LOGGER = logging.getLogger(__name__)
+_KODI_DEVICES = dict()
+
+
+def setup(hass, config):
+    """ Setup the Kodi component. """
+
+    for config_key in extract_domain_configs(config, DOMAIN):
+        add_kodi_device(config[config_key])
+
```

Comment:
Can you add discovery functionality so that the kodi media player and notify platforms will be setup automatically when this component is loaded?